<robot name="kirin" xmlns:xacro="http://ros.org/wiki/xacro">

  <material name="default">
    <color rgba="0.7 0.7 0.7 1.0"/> 
  </material>
  

  <!-- link settings -->
  <link name="fix_base">
    <visual name="">
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 ${pi/2}"/>
      <geometry>
        <mesh filename="package://kirin/resources/light/fix.stl"/>
      </geometry>
      <material name="default"/>
    </visual>
  </link>

  <link name="theta_link">
    <visual name="">
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 ${pi/2}"/>
      <geometry>
        <mesh filename="package://kirin/resources/light/theta.stl"/>
      </geometry>
      <material name="default"/>
    </visual>
  </link>

  <link name="z_link">
    <visual name="">
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 ${pi/2}"/>
      <geometry>
        <mesh filename="package://kirin/resources/light/z.stl"/>
      </geometry>
      <material name="default"/>
    </visual>
  </link>

  <link name="r_link">
    <visual name="">
      <origin xyz="0.1725 0.0 0.0" rpy="0.0 0.0 ${pi/2}"/>
      <geometry>
        <mesh filename="package://kirin/resources/light/r.stl"/>
      </geometry>
      <material name="default"/>
    </visual>
  </link>

  <link name="phi_link"/>
  <link name="depart"/>

  <!-- joint settings -->
  <joint name="theta_joint" type="continuous">
    <parent link="fix_base"/>
    <child link="theta_link"/>
    <origin xyz="0.0 0.0 0.022" rpy="0.0 0.0 0.0"/>
    <axis xyz="0.0 0.0 1.0"/>
  </joint>

  <joint name="z_joint" type="prismatic">
    <parent link="theta_link"/>
    <child link="z_link"/>
    <origin xyz="0.0 0.0 0.128" rpy="0.0 0.0 0.0"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit lower="0.0" upper="0.15" effort="100" velocity="2.0"/>
  </joint>

  <joint name="r_joint" type="prismatic">
    <parent link="z_link"/>
    <child link="r_link"/>
    <origin xyz="0.201 0.0 0.0" rpy="0.0 0.0 0.0"/>
    <axis xyz="1.0 0.0 0.0"/>
    <limit lower="0.0" upper="0.413" effort="100" velocity="5.0"/>
  </joint>

  <joint name="phi_joint" type="revolute">
    <parent link="r_link"/>
    <child link="phi_link"/>
    <origin xyz="0.345 0.0 -0.0235" rpy="0.0 0.0 0.0"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit lower="${-pi}" upper="${pi}" effort="10.0" velocity="${10*pi}"/>
  </joint>

  <joint name="depart_joint" type="fixed">
    <parent link="fix_base"/>
    <child link="depart"/>
    <origin xyz="0.55 0.55 0.225" rpy="0.0 0.0 ${-pi/2}"/>
  </joint>

</robot>
